var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -362,6 +363,18 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     conflicts("+stokhos", when="%xl")
     conflicts("+stokhos", when="%xl_r")
 
+    # Current Windows support, only have serial static builds
+    conflicts(
+        "+shared",
+        when="platform=windows",
+        msg="Only serial static builds are supported on Windows currently.",
```

Comment:
Customize each message to the variant, e.g.:

```
Only static builds are supported on Windows currently
```

somewhat repetitious, but that allows us to address each of these in potentially separate PRs without the message becoming wrong.